test/06-generic.js

```diff
@@ -33,6 +33,12 @@ tape('06 generic type', function (t) {
   t.equal(encoded[8], 0, 'falsy number as an array item') // isn't this the same as "numbers no encoded"?
   t.deepEqual(encoded[9], Buffer.from([3, 2, 1]), 'buffer untouched')
 
-  t.deepEquals(bfe.decode(encoded), values, 'decode works')
+  // WARNING! a buffer is encoded as a buffer, which can lead to strange behaviour on decoding!!
+  //
```

Comment:
@mixmix I went hunting what happened to the tests that should have failed (see https://github.com/ssb-ngi-pointer/ssb-bfe/pull/22#issuecomment-896745637), and realized it was this `encode.slice(0, encoded.length - 1)`. The slice removes the last array item, which is the 030201 test buffer. It basically opted-out of testing that buffers are encodable/decodable (yes, they may fail 10% of the time).

This is one of those cases I would never have merged or approved the changes to this test file, because it removes an important assertion (the assertion **is** correct, even if the _implementation_ may be incorrect!). Just as a concrete example that trivial changes should not be merged without a review from the authors, because it may not be trivial after all.

Reply:
Thanks for supporting this. Yeah I remember this now.